### PR TITLE
RHDEVDOCS-5221: Overview of Pipelines

### PR DESCRIPTION
**Pull request by mramendi**
Version(s):

5.12, 5.13, 5.14

Issue:

https://issues.redhat.com/browse/RHDEVDOCS-5221

Link to docs preview:

https://59556--docspreview.netlify.app/openshift-enterprise/latest/cicd/pipelines/understanding-openshift-pipelines.html

(Only the "Overview of Red Hat OpenShift Pipelines" section is added, other content is not updated)

QE review:
- [x] QE has approved this change.

Additional information:

This PR adds an initial brief overview of OpenShift Pipelines. This should later serve as the baseline for restructuring Pipelines documentation. For now, no resructuring is performed and the overview remains under the existing "Understanding..." header.

### CODE REVIEW

File: modules/op-overview.adoc
Comment (khrm) [inline modules/op-overview.adoc:39]: Should we can add the interceptor and it's brief details also?

Comment (koustavsaha) [reply modules/op-overview.adoc:39]: Agreed

File: modules/op-overview.adoc
Comment (koustavsaha) [inline modules/op-overview.adoc:52]: We should mention Tekton Hub even though it will be deprecated in future

**Review by koustavsaha — CHANGES_REQUESTED** [review]

**Review by concaf — APPROVED** [review]

File: modules/op-overview.adoc
Comment (mramendi) [reply modules/op-overview.adoc]: Replaced with "definition"

**Review by vdemeester — APPROVED** [review]

**Review by ppitonak — APPROVED** [review]

File: cicd/pipelines/understanding-openshift-pipelines.adoc
Comment (abrennan89) [inline cicd/pipelines/understanding-openshift-pipelines.adoc:23]: Per the OCP doc guidelines, "overview" shouldn't be used in headings: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assemblymodule-titles-and-section-headings

Maybe you could combine this with the content above the "Key features" heading rather than needing it to be a separate module?

File: modules/op-overview.adoc
Comment (abrennan89) [inline modules/op-overview.adoc:15]: All h2 headings should have anchors per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-in-module-files

File: modules/op-overview.adoc
Comment (abrennan89) [inline modules/op-overview.adoc:11]: OpenShift and other product titles should use the attributes from the common attributes file: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#product-title-and-version

Comment (abrennan89) [reply modules/op-overview.adoc:11]: I also think that the "(Kubernetes)" can be omitted here as assumed knowledge since this is explained as part of core OCP docs

**Review by abrennan89 — COMMENTED** [review]
Left a couple of initial comments, will take another look once those are addressed if that's OK 🙂

File: cicd/pipelines/understanding-openshift-pipelines.adoc
Comment (mramendi) [reply cicd/pipelines/understanding-openshift-pipelines.adoc:23]: "Key features" is a bullet list format (in the assembly, before modules) common to several assemblies the existing doc structure, and I don't want to touch it before i can restructure. I was not aware of the "overview" ban (which makes no sense to me) and I'll try to conjure a replacement.

File: modules/op-overview.adoc
Comment (mramendi) [reply modules/op-overview.adoc:11]: I will remove Kubernetes. The problem in this specific context is that "OpenShift resource" is the description of a resource as opposed to a product name. It's never "OpenSnift Container Platform resource" or "OKD resource" or even "Red Hat OpenShift resource" - it is always "OpenShift resource". I was not able to find a suitable attribute. If there is a suitable attribute please point it out.

File: modules/op-overview.adoc
Comment (mramendi) [reply modules/op-overview.adoc:15]: Will do.

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
Hi @mramendi: Nice work! PTAL at the suggestions and queries.

Incorporate whatever you can and the rest you can create a JIRA and carry over the work to come back and handle after the GA.

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:18]: I intentionally want to leave this ambiguous as the brief overview is not the place to discuss best design practice. I will reword to make the ambiguity clearer.

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:13]: "Components" also refers to things like Resolvers and Results. It's not a great heading. It's beacuse "Overview" is banned and "Concepts" is already uses and I have to have something for 1.11 (for 1.12 I hope to rework the entire assembly).

"Building blocks" or just "blocks" are only pipelines and tasks

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:18]: see above

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:11]: I don't understand removal of the variable here.

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:20]: I am intentionally describing concepts not just resources that represent them. I tried to clarify this.

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:16]: A big rewrite is not a possibility in this timeframe - it was hard enough to get this one agreed.

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:43]: Again the header is about concepts not just resources

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:52]: No. An Event Listener is not a resource as far as I am aware.

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:113]: Well, maybe you'll like the new one better - not really sure how to make this idea clearer.

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:68]: "use by using"?

File: modules/op-components.adoc
Comment (Srivaralakshmi) [reply modules/op-components.adoc]: If you can use reusable resolvers and they are components then use the word components throughout. That makes more sense.

File: modules/op-components.adoc
Comment (Srivaralakshmi) [reply modules/op-components.adoc]: Overview when it must be simple it must also be clear with no ambiguity. Thanks for rewording.

File: modules/op-components.adoc
Comment (Srivaralakshmi) [inline modules/op-components.adoc:18]: Inconsistency in the article usage. LGTM, trusting you will correct this.

File: modules/op-components.adoc
Comment (Srivaralakshmi) [reply modules/op-components.adoc]: @mramendi please retain the variable..  my bad! it is a typo when I copied it from my text pad as I was trying to reword it.

File: modules/op-components.adoc
Comment (Srivaralakshmi) [inline modules/op-components.adoc:20]:
```suggestion
A _task_, represented by a `Task` resource, is a collection of steps, which always run in a fixed sequential order. {pipelines-title} runs all steps of a task as containers in the same pod. This pod provides an environment that enables steps to access shared resources. For example, you can mount a volume in a task, and each step of the task can access this volume.
```

File: modules/op-components.adoc
Comment (Srivaralakshmi) [inline modules/op-components.adoc:66]:
```suggestion
With _resolvers_, you can retrieve tasks and pipelines from a repository by using an identifier. You can reference a task or pipeline by using a resolver instead of defining the task or pipeline in your CRs. When the task or pipeline must be run, {pipelines-shortname} retrieves and runs the task or pipeline.
```
Instead of defining the task or pipeline in your CRs, are you referencing a task or a pipeline **that is** using a resolver, or are you referencing a task or a pipeline **by using** a resolver? Do you see how the writing conveys different meanings and is resulting in ambiguity and confusion for the reader?? 

Read both sentences carefully, understand what we need to exactly convey to the reader, and accordingly, rewrite for clarity.

File: modules/op-components.adoc
Comment (Srivaralakshmi) [inline modules/op-components.adoc:70]:
```suggestion
* _Hub resolver_: Retrieves a task or pipeline from the Pipelines Catalog available on the Artifact Hub or Tekton Hub.
```
Do not miss **the period**.

File: modules/op-components.adoc
Comment (Srivaralakshmi) [inline modules/op-components.adoc:72]:
```suggestion
* _Bundles resolver_: Retrieves a task or pipeline from a Tekton bundle, which is an Open Container Initiative (OCI) image available from any OCI repository including an OpenShift image registry.
```
expand OCI.

File: modules/op-components.adoc
Comment (Srivaralakshmi) [inline modules/op-components.adoc:90]: @mramendi [Same comment](https://github.com/openshift/openshift-docs/pull/59556#discussion_r1228061710) again. what is actually signed by a key -- storage or data?
I am sure it is the data here. Be consistent with the words you use, so the reader does not get confused.

See the following lines written by you: "The full execution data of every task run and pipeline run, including run specs and logs, **is stored** in the `TaskRun` and `PipelineRun` CRs and **can be signed and stored** in a repository with {tekton-chains}."

File: modules/op-components.adoc
Comment (Srivaralakshmi) [inline modules/op-components.adoc:113]: I love the change and rewrite. Super Misha.
Keep this together. Else, when you say 'this` definition, continuity is lost for the reader.

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@mramendi Thanks for the changes and rewrite. Lovely work!

Some minor suggestions and queries.
Incorporate whatever you can and the rest you can create a JIRA and carry over the work to come back and handle after the GA.

Please feel free to get another quick PR review for a quick approval/lgtm with whoever is available from the review team.
Thank you!

File: modules/op-components.adoc
Comment (bburt-rh) [inline modules/op-components.adoc:18]:
```suggestion
The smallest block of the Pipelines CI/CD process is a _step_. A step is a single operation, such as running one or several unit tests for a Python web application or a Java program compilation. {pipelines-title} performs each step in a separate container.
```

File: modules/op-components.adoc
Comment (mramendi) [reply modules/op-components.adoc:24]: Definitely not. The task can have parameters and a pipeline sets them; the pipeline itself has different parameters. The real problem here is that the pipeline run sets the parameter values for a pipeline but the pipeline run is not yet defined in the text. I'll try to write around this though.

File: modules/op-components.adoc
Comment (Srivaralakshmi) [inline modules/op-components.adoc:45]: Love this!! thank you!

File: modules/op-components.adoc
Comment (Srivaralakshmi) [inline modules/op-components.adoc:99]:
```suggestion
To monitor and mitigate any possible security issues, {pipelines-shortname} provides {tekton-chains}, a method of storing the entire content of every task run and pipeline run, including any data (such as mounted volumes) processed in the run. The stored information is signed by a key so that any tampering with the information is evident.
```

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@mramendi Thanks for your patience and time in taking care of all the suggested changes. Amazing work!!

Just some minor comments to polish some nitpick. Rest LGTM!

**Review by Srivaralakshmi — APPROVED** [review]
@mramendi Excellent work! Thanks for all the changes. Approved! LGTM.

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@mramendi Sorry to have missed this one comment. We only have an attribute for Tekton hub and NOT artifact hub. You need to create one. Also, wherever Tekton hub and Artifact hub is appearing in the PR ensure that you use attributes for them and they are appearing clean in the preview.

**Review by Srivaralakshmi — APPROVED** [review]
@mramendi LGTM and Approved. Thanks for all the hard work! I did not want to block this PR past the release date.

But, this assembly needs a heavy restructuring. The concepts must go as individual modules with their headings/anchor titles changed appropriately. Can you implement the changes suggested by @abrennan89  to have a clean PR? It will be great!  I do not know the urgency and level of changes you have in mind for the PR. I leave the decision to you.

File: modules/op-components.adoc
Comment (joaedwar) [inline modules/op-components.adoc:25]: This is a great explanation, but I am not sure it is fully reflected in the diagram above. Is it possible to speak with Jess Schaefer about making this a bit clearer - so that it is clear that both tasks have to complete, to start task D. Also , what happens if just on task completes but the other does not ?

File: modules/op-components.adoc
Comment (joaedwar) [inline modules/op-components.adoc:43]: Is it possible to present these a bit differently...

- Parameter as a pipeline ::
- Parameter as a task::

File: modules/op-components.adoc
Comment (joaedwar) [inline modules/op-components.adoc:64]: How come we do not use catalog for sentence case?

**Pull request closed without merging**
